src/MSALWrapper.Test/AuthFlow/AuthFlowExecutorTest.cs

```diff
@@ -0,0 +1,623 @@
+    using System.Threading.Tasks;
+    using FluentAssertions;
+    using Microsoft.Authentication.MSALWrapper;
+    using Microsoft.Authentication.MSALWrapper.AuthFlow;
+    using Microsoft.Authentication.MSALWrapper.Test;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Identity.Client;
+    using Microsoft.IdentityModel.JsonWebTokens;
+    using Moq;
+    using NLog.Extensions.Logging;
+    using NLog.Targets;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// The main auth flow test.
+    /// </summary>
+    public class AuthFlowExecutorTest
+    {
+        private IServiceProvider serviceProvider;
+        private MemoryTarget logTarget;
+        private TokenResult tokenResult;
+        private IEnumerable<IAuthFlow> authFlows;
+
+        /// <summary>
+        /// The test setup.
+        /// </summary>
+        [SetUp]
+        public void Setup()
+        {
+            // Setup in memory logging target with NLog - allows making assertions against what has been logged.
+            var loggingConfig = new NLog.Config.LoggingConfiguration();
+            this.logTarget = new MemoryTarget("memory_target");
+            loggingConfig.AddTarget(this.logTarget);
+            loggingConfig.AddRuleForAllLevels(this.logTarget);
+            this.authFlows = new List<IAuthFlow>();
+
+            // Setup Dependency Injection container to provide logger and out class under test (the "subject")
+            this.serviceProvider = new ServiceCollection()
+             .AddLogging(loggingBuilder =>
+             {
+                 // configure Logging with NLog
+                 loggingBuilder.ClearProviders();
+                 loggingBuilder.SetMinimumLevel(Microsoft.Extensions.Logging.LogLevel.Trace);
+                 loggingBuilder.AddNLog(loggingConfig);
+             })
+             .BuildServiceProvider();
+
+            // Mock successful token result
+            this.tokenResult = new TokenResult(new JsonWebToken(TokenResultTest.FakeToken));
+        }
+
+        [Test]
+        public void ConstructorWith_BothNullArgs()
+        {
+            Action authFlowExecutor = () => new AuthFlowExecutor(null, null);
+
+            // Assert
+            authFlowExecutor.Should().Throw<ArgumentNullException>();
+        }
+
+        [Test]
+        public void ConstructorWith_Null_Logger()
+        {
+            Action authFlowExecutor = () => new AuthFlowExecutor(null, this.authFlows);
+
+            // Assert
+            authFlowExecutor.Should().Throw<ArgumentNullException>();
+        }
+
+        [Test]
+        public void ConstructorWith_Null_AuthFlows()
+        {
+            var logger = this.serviceProvider.GetService<ILogger<AuthFlowExecutor>>();
+            Action authFlowExecutor = () => new AuthFlowExecutor(logger, null);
+
+            // Assert
+            authFlowExecutor.Should().Throw<ArgumentNullException>();
+        }

```

Comment:
We don't appear to accept any combination of nulls here because we do
```csharp
this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
this.authflows = authFlows ?? throw new ArgumentNullException(nameof(authFlows));
```
These tests would maybe be necessary in Python, but can't we crank up the strictness of the compiler and disallow nulls here?

Seems to me like in C# we should be able to rely on the compiler to do this kind of testing and save us from writing the null checking code above. Maybe I have the wrong idea though.

Reply:
I think you have the right idea @reillysiemens but we have to turn on no nulls for the entire project. I think that could be it's own small feature.

Reply:
I do have tests for null checks in the constructor. Kyle wanted me to catch and throw immediately if we get a null logger or auth flows because that means an error on the caller's side.

Reply:
If we don't have an issue for disabling nulls project-wide we should do that. I don't know the best way to capture things like this though where the ramification would be "here's a whole slew of tests you can just delete".

Reply:
@shalinikhare27: I think you've done the right thing here given the current configuration of our project. I'm going to resolve this so you're not blocked on it. The point I was making is that in the future we should be able to just avoid writing these kinds of tests altogether and rely on the compiler. 😄